UserGuide: update degree planner details and FAQ

Letm3through opened the pull request:
Let's update the user commands related to the degree planner and update FAQ
so that users are able to better understand the commands.

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -237,30 +237,31 @@ Examples:
 Deletes an existing module from the degree planner. +
 Format: `planner_delete CODE`
 
 * `CODE` indicates module code.
```
Let's remove this line since this was covered in the section above (add module to degree planner)

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -237,30 +237,31 @@ Examples:
-* After adding, the changes will be reflected in the degree planner accordingly.
+* `SEMESTER` indicates the semester.
+* `YEAR` and `SEMESTER` must be either 1, 2, 3 or 4.
+* 3 or 4 for `SEMESTER` indicates special semester 1 and special semester 2 respectively.
```
Let's remove these few lines since these parameters were covered in a previous section (add module to degree planner).

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
+=== Listing degree planners: `planner_list_all` `[coming in v2.0]`
+
+Shows a list of all degree planners. +
+Formats: `planner_list_all`
```
`Formats` -> `Format`

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
+=== Listing a degree planner: `planner_list` `[coming in v2.0]`
+
+Shows a list of a specific degree planner. +
+Formats: `planner_list y/YEAR s/SEMESTER`
```
`Formats` -> `Format`

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
+* `YEAR` indicates the year.
+* `SEMESTER` indicates the semester.
+* `YEAR` and `SEMESTER` must be either 1, 2, 3 or 4.
+* 3 or 4 for `SEMESTER` indicates special semester 1 and special semester 2 respectively.
```
Let's remove these few lines since these parameters were covered in a previous section (add module to degree planner).

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
+* `planner_list y/1 s/2` +
+Shows a degree planner for year 1 semester 2
 
-Displays a list of modules that can be taken in the current selected semester.
```
Currently, there is no way to `select` a semester. 
I think for now, we can phrase it as:
```
Displays a list of modules that can be added to the degree planner.
```

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -237,30 +237,31 @@ Examples:
 Format: `planner_delete CODE`
 
 * `CODE` indicates module code.
+* After deleting, the changes will be reflected in the degree planner accordingly.
```
...And refine this sentence now that the previous line is deleted.

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
-* The modules suggested are not currently in the planner.
+Displays a list of modules that can be taken in the current selected academic semester.
+
+* The modules suggested are currently not found in the degree planner.
```
I think you can remove this sentence out.

Creastery commented on docs/UserGuide.adoc:
```diff
@@ -269,13 +270,35 @@ Examples:
+* `YEAR` indicates the year.
+* `SEMESTER` indicates the semester.
+* `YEAR` and `SEMESTER` must be either 1, 2, 3 or 4.
+** 3 or 4 for `SEMESTER` indicates special semesters 1 and 2 respectively.
```
These lines can be removed since the parameters are already explained in "add a module to degree planner" already.

Letm3through committed:
UserGuide: update degree planner details and FAQ

Let's update the user commands related to the degree planner and update FAQ
so that users are able to better understand the commands.

Creastery reviewed (approved):
LGTM. Approving and merging.
